.github/workflows/run-test-n-times.yml

```diff
@@ -0,0 +1,141 @@
+name: Run Test N-times
+
+on:
+  workflow_dispatch:
+    inputs:
+      commit:
+        description: "Commit SHA"
+        required: true
+      n_runs:
+        description: "Number of times to repeat the test"
+        required: true
+      test_name:
+        description: "Name of the test suite to run, include -testify.m flag here if desired (i.e. 'TestAcquireShard_DeadlineExceededErrorSuite' or 'TestFunctionalSuite -testify.m=TestUpdateWorkflow')"
+        required: true
+
+env:
+  COMMIT: ${{ github.event.inputs.commit || github.sha }}
+  DOCKER_COMPOSE_FILE: ./develop/github/docker-compose.yml
+  TEMPORAL_VERSION_CHECK_DISABLED: 1
+  BUILDKITE_ANALYTICS_TOKEN: ${{ secrets.BUILDKITE_ANALYTICS_TOKEN }}
+
+jobs:
+  misc-checks:
+    name: Misc checks
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [ubuntu-20.04]
+    runs-on: ${{ matrix.runs-on }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          ref: ${{ env.COMMIT }}
+          submodules: true
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+          check-latest: true
+
+      - uses: arduino/setup-protoc@v3
+
+      - run: make ci-build-misc
+
+      - run: make build-tests
+
+  cache-docker-images:
+    name: Cache Docker images
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [ubuntu-20.04]
+    runs-on: ${{ matrix.runs-on }}
+    steps:
+      - uses: ScribeMD/docker-cache@0.3.7
+        with:
+          key: docker-${{ runner.os }}-${{ hashFiles(env.DOCKER_COMPOSE_FILE) }}
+
+      - uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          ref: ${{ env.COMMIT }}
+
+      - run: docker compose -f ${{ env.DOCKER_COMPOSE_FILE }} pull
+
+  functional-test-n-times:
+    name: Functional test n-times
+    needs: [misc-checks, cache-docker-images]
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [ubuntu-20.04]
+        name: [cass_es, cass_es8, sqlite, mysql8, postgres12, postgres12_pgx]
+        shard_index: [0, 1, 2]
```

Comment:
We don't need shards if we're just running one test.

But in general, this seems like it's going to get out of sync with the main test workflow. Can we consolidate things into one workflow, i.e. add optional test name and count to the main run-tests?

Reply:
yes, I totally agree about getting out of sync being an issue.
Do you have any ideas for how we can conditionally not use the shards if we detect that the input test flags say to just use one test?

Reply:
the coverage report also doesn't make sense when looking at just one test (you end up with a really low coverage percent, unless you also change the coverage package scope), which is why I created a new make target instead of re-using the existing `functional-test-coverage` target